### PR TITLE
INS-32: reference Domain tests added

### PR DESCRIPTION
**Pull request by ilyapt**
**- What I did**
reference Domain tests added

### CODE REVIEW

File: genesis/public/core/reference_test.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import (
 	"github.com/insolar/insolar/genesis/model/domain"
 	"github.com/insolar/insolar/genesis/model/object"
 	"github.com/stretchr/testify/assert"
+	"github.com/insolar/insolar/genesis/model/class"
```
Comment (dmitry-panchenko) [inline genesis/public/core/reference_test.go]: Are imports correctly sorted?

File: genesis/public/core/reference_test.go [hunk 2]
```diff
@@ -82,3 +84,146 @@ func TestNewReferenceDomain_WithNoParent(t *testing.T) {
...
+	refDomain := newReferenceDomain(nil)
+	domainID := refDomain.GetClassID()
+	assert.Equal(t, class.ReferenceDomainID, domainID)
+}
+
+func TestReferenceDomain_RegisterReference(t *testing.T) {
+	domain := "1"
+	record := "1"
+	refObject, err := object.NewReference(record, domain, object.GlobalScope)
+	assert.Nil(t, err)
```
Comment (dmitry-panchenko) [inline genesis/public/core/reference_test.go]: You could you `assert.NoError` here and in similar checks below

**Comment by ilyapt** [comment]
All issues fixed

File: genesis/public/core/reference_test.go [hunk 3]
```diff
@@ -81,3 +83,146 @@ func TestNewReferenceDomain_WithNoParent(t *testing.T) {
...
+	assert.NoError(t, err)
+	assert.Equal(t, domain, resolved.Domain)
+	assert.Equal(t, record, resolved.Record)
+
+}
+
+func TestReferenceDomain_ResolveReference_IncorrectRef(t *testing.T) {
+	refDomain := newReferenceDomain(nil)
+	_, err := refDomain.ResolveReference("1")
+	assert.Equal(t, "object with record 1 does not exist", err.Error())
```
Comment (dmitry-panchenko) [inline genesis/public/core/reference_test.go]: You could use `assert.EqualError` for error checks.

`assert.EqualError(t, err, "object with record 1 does not exist")`

Comment (ilyapt) [reply genesis/public/core/reference_test.go]: Thx, fixed

**Pull request merged**